src/periphery/MagicApeLens.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "interfaces/IApeCoinStaking.sol";
+
+contract MagicApeLens {
+    uint256 constant PRECISION = 1e18;
+    uint256 constant BPS_PRECISION = 1e4;
+    address public immutable APE_COIN_STAKING_CONTRACT = 0x5954aB967Bc958940b7EB73ee84797Dc8a2AFbb9;
+    address public immutable APE_COIN_CONTRACT = 0x4d224452801ACEd8B2F0aebE155379bb5D594381;
+
+    struct PoolInfo {
+        uint256 apr;
+        uint256 stakedAmount;
+        uint256 poolRewardsPerHour;
+        uint256 poolRewardsPerDay;
+        uint256 rewardPerHour;
+        uint256 poolRewardsPerTokenPerDay;
+    }
+
+    function getApeCoinInfo() external view returns (PoolInfo memory info) {
+        (IApeCoinStaking.PoolUI memory apeCoinInfo, , , ) = IApeCoinStaking(APE_COIN_STAKING_CONTRACT).getPoolsUI();
+        return getPoolInfo(apeCoinInfo);
+    }
+
+    function getBAYCInfo() external view returns (PoolInfo memory info) {
```

Comment:
we don't need these tho as we are only staking ape, but I don't mind having it there

Reply:
I figured we might do something related to this with Potion, so I added.